src/state/translations-mixin.ts

```diff
@@ -58,18 +69,33 @@ export default <T extends Constructor<HassBaseEl>>(superClass: T) =>
       }
       const oldHass = changedProps.get("hass");
       if (this.hass?.panels && oldHass.panels !== this.hass.panels) {
-        this._loadFragmentTranslations(this.hass.language, this.hass.panelUrl);
+        this._loadFragmentTranslations(
+          this.hass.locale?.language,
+          this.hass.panelUrl
+        );
       }
     }
 
-    protected hassConnected() {
+    protected async hassConnected() {
       super.hassConnected();
-      getUserLanguage(this.hass!).then((language) => {
-        if (language && this.hass!.language !== language) {
-          // We just get language from backend, no need to save back
-          this._selectLanguage(language, false);
-        }
-      });
```

Comment:
We should keep this logic as it was, it should just return `FrontendTranslationData` instead of just the language.